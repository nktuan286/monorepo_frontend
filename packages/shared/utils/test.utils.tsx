import { ReactNode, ReactElement } from 'react';
import {
  render, RenderOptions,
} from '@testing-library/react';
import { HashRouter } from 'react-router-dom';
import { Theme } from '../src/styling';

function WrapedRender({ children }: {children: ReactNode}) {
  return (
    <HashRouter>
      <Theme>
        {children}
      </Theme>
    </HashRouter>
  );
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>,
) => render(ui, { wrapper: WrapedRender, ...options });

export * from '@testing-library/react';
export { customRender as render };
