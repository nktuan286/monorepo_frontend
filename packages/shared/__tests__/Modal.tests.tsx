/* eslint-disable react/no-children-prop */
import { render } from '../utils/test.utils';
import 'jest-styled-components';
import {
  Modal,
} from '../src/components/Modal';

function Component({ isOpen = false }: { isOpen?: boolean }) {
  return (
    <Modal title="Modal title" isOpen={isOpen}>
      This is content of modal
    </Modal>
  );
}

const renderModal = (isOpen: boolean = true) => {
  const { container, rerender } = render(
    <Component isOpen={isOpen} />,
  );

  const rerenderForm = () => {
    rerender(
      <Component isOpen={isOpen} />,
    );
  };

  return { container, rerenderForm };
};

describe('<Modal /> Tests', () => {
  test('Render <Modal /> with out crash', () => {
    const { container } = renderModal();
    expect(container).toMatchSnapshot();
  });
});
