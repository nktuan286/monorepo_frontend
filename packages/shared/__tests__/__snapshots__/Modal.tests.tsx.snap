// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> Tests Render <Modal /> with out crash 1`] = `
<div>
  <div
    class="sc-gEvEer sc-iGgWBj esOBjJ eKTvWW"
  >
    <a
      class="sc-gsFSXq egIMRp"
      id="btn-close-modal"
    />
    <div
      class="sc-gEvEer sc-eqUAAy sc-jXbUNg esOBjJ bWNTFc dWQDrZ"
    >
      <div
        aria-labelledby="modal-title"
        aria-live="assertive"
        aria-modal="true"
        class="sc-dhKdcB glQfRO"
      >
        <span
          class="sc-kpDqfm fBVlTq"
          id="modal-title"
        >
          Modal title
        </span>
        <a
          aria-label="Close"
          class="sc-kAyceB hcSyvB"
        >
          <i
            class="sc-dcJsrY sc-imWYAI bxoAal ipvehj mi-close"
          />
        </a>
      </div>
      This is content of modal
    </div>
  </div>
</div>
`;
